src/test/kotlin/com/example/Employee/services/EmployeeServiceTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.Employee.services
+
+
+import com.example.Employee.exceptions.EntityNotFoundException
+import com.example.Employee.exceptions.InvalidRequestException
+import com.example.Employee.models.Employee
+import com.example.Employee.repositories.EmployeeRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import java.util.*
+import org.amshove.kluent.`should be equal to`
+import org.amshove.kluent.invoking
+import org.amshove.kluent.shouldThrow
+import org.junit.jupiter.api.Assertions.*
+
+internal class EmployeeServiceTest {
+    private val employeeRepository= mockk<EmployeeRepository>()
+    private val employeeService= EmployeeService(employeeRepository)
+
+    @Test
+    fun `should create employee`() {
+        every {employeeRepository.save(fakeEmployee)} returns fakeEmployee.copy(employeeId=1)
+
+        employeeService.addDetails(fakeEmployee)
+
+        verify { employeeRepository.save(fakeEmployee)}
+    }
+
+    @Test
+    fun `should not create employee when age is less than 18`(){
+        //val employeeToCreate = Employee(employeeId = 0, name= "Vikram",age=16,address="Hyd",designation="Intern",department="LPD", location = "Bangalore", aadharNumber = "5678678678", pancard = "KYP678")
+        val employeeToCreate = fakeEmployee.copy(age=16)
+        invoking { employeeService.addDetails(employeeToCreate) } shouldThrow InvalidRequestException("Age should be above 18")
+
+        verify(exactly = 0){employeeRepository.save(any())}
+    }
+
+    @Test
+    fun `should get all the employees`(){
+        every {employeeRepository.findAll()} returns listOf(fakeEmployee)
+
+        val fetchedEmployees = employeeService.get()
+
+        fetchedEmployees `should be equal to` listOf(fakeEmployee)
+    }
+
+    @Test
+    fun `should get the employee`(){
+        every {employeeRepository.findById(fakeEmployee.employeeId)} returns Optional.of(fakeEmployee)
+
+        val fetchedEmployee= employeeService.getEmployee(employeeId = fakeEmployee.employeeId)
+
+        fetchedEmployee `should be equal to` fakeEmployee
+    }
+
+    @Test
+    fun `should throw error for non existent employee` (){
+        every { employeeRepository.findById(fakeEmployee.employeeId) } returns Optional.empty()
+
+        invoking { employeeService.getEmployee(fakeEmployee.employeeId) } shouldThrow EntityNotFoundException("employee not found")
+    }
+
+    @Test
+    fun `should delete employee` (){
+        every { employeeRepository.findById(fakeEmployee.employeeId) } returns Optional.of(fakeEmployee)
+        every { employeeRepository.deleteById(fakeEmployee.employeeId)} returns Unit
```

Comment:
Returning Unit is the default value for any stub.

Reply:
But if we won't write it there,  we will get an error saying no answer provided for employeeRepository.deleteById() right?

Reply:
Test it without the stub. If it throws an error, have the stub.